frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react'
+import { Button, Form } from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+
+
+function PersonalScheduleForm({ initialPersonalSchedule, submitAction, buttonLabel = "Create" }) {
+
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialPersonalSchedule || {}, }
+    );
+    // Stryker enable all
+
+    const navigate = useNavigate();
+
+    return (
+
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialPersonalSchedule && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid="PersonalScheduleForm-id"
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialPersonalSchedule.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="name">Name</Form.Label>
+                <Form.Control
+                    data-testid="PersonalScheduleForm-name"
+                    id="name"
+                    type="text"
+                    isInvalid={Boolean(errors.name)}
+                    {...register("name", {
+                        required: "Name is required."
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.name?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="description">Description</Form.Label>
+                <Form.Control
+                    data-testid="PersonalScheduleForm-description"
+                    id="description"
+                    type="text"
+                    isInvalid={Boolean(errors.description)}
+                    {...register("description", {
```

Comment:
I wonder whether it would be helpful to use the QuarterSelectorDropdown here.   I think that would result in fewer errors in terms of data entry.

Reply:
### New Commit:
Finished - Replaced type text form control with the SingleQuarterDropdown component. Ranges from Winter 2008 to Fall 2024 right now. 
### Future Considerations:
 Future fixes may be to edit the range if necessary, and also the SingleQuarterDropdown component starts from the earliest quarter and ends at the most recent, which is annoying and could also be a future issue to resolve. 
![image](https://user-images.githubusercontent.com/91104605/157344162-383ab5ed-cd39-4cc8-9e52-044244413903.png)